t/24-worker-job.t

```diff
@@ -460,4 +461,84 @@ subtest 'handling API failures' => sub {
       or diag explain \@uploaded_assets;
 };
 
+# reset state
+@happended_events = ();
+$isotovideo->is_running(1);
+
+# exercise another job live-cycle simulating an upload failure
+subtest 'handle upload failure' => sub {
+    # assume all uploads fail (set this temporarily to 1 to "test the test" which should fail then)
+    $upload_result = 0;
+
+    # create and accept new job
+    $job = OpenQA::Worker::Job->new($worker, $client, {id => 4, URL => 'url'});
+    $job->on(status_changed => $handle_job_event);
+    $job->accept;
+
+    combined_like(
+        sub {
+            $job->start();
+            wait_until_upload_concluded($job);
+        },
+        qr/isotovideo has been started/,
+        'isotovideo startup logged'
+    );
+
+    # assume isotovideo generated some logs
+    my $log_dir = $pool_directory->child('ulogs');
+    $log_dir->make_path;
+    $log_dir->child('foo')->spurt('some log');
+    $log_dir->child('bar')->spurt('another log');
+
+    # assume isotovideo generated some assets
+    my $asset_dir = $pool_directory->child('assets_public');
+    $asset_dir->make_path;
+    $asset_dir->child('hdd1.qcow')->spurt('data');
+    $asset_dir->child('hdd2.qcow')->spurt('more data');
+
+    combined_like(
+        sub {
+            $job->stop('done');
+            wait_until_upload_concluded($job);
+        },
+        qr/Result: done/,
+        'job stopped after done'
+    );
+
+    is($client->register_called, 1, 'worker tries to register itself again after an API failure');
```

Comment:
Just a minor point, but `worker tried to register itself again [...]` is probably better?

Reply:
And in fact is should be `worker tried to register itself again after an upload failure`. I also forgot to unset `$client->register_called`.